From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eric Rahm <ericrahm@gmail.com>
Date: Thu, 13 Jan 2022 10:10:31 -0800
Subject: fix: Always bypass extra authenticator dialogs

This fixes two issues:
  1. Looking up a non-existent pref branch leads to a crash.
  2. An additional auth dialog shows up if we bypass that.

Now the auth dialog is perma-bypassed so that we don't block completion.

diff --git a/chrome/browser/webauthn/chrome_authenticator_request_delegate.cc b/chrome/browser/webauthn/chrome_authenticator_request_delegate.cc
index 1ff048c280d7e77b569eb57831bfab8ba4007f28..09ad06f036cb33a5b4a857818e49992b5258b8f2 100644
--- a/chrome/browser/webauthn/chrome_authenticator_request_delegate.cc
+++ b/chrome/browser/webauthn/chrome_authenticator_request_delegate.cc
@@ -76,6 +76,7 @@ ChromeAuthenticatorRequestDelegate::TestObserver* g_observer = nullptr;
 bool IsWebauthnRPIDListedInEnterprisePolicy(
     content::BrowserContext* browser_context,
     const std::string& relying_party_id) {
+#if 0
   const auto* profile = static_cast<ElectronBrowserContext*>(browser_context);
   const PrefService* prefs = profile->prefs();
   const base::ListValue* permit_attestation =
@@ -85,6 +86,10 @@ bool IsWebauthnRPIDListedInEnterprisePolicy(
                      [&relying_party_id](const base::Value& v) {
                        return v.GetString() == relying_party_id;
                      });
+#endif
+
+  // TODO(eric): once modal dialogs are displayed we can make this false.
+  return true;
 }
 
 std::string Base64(base::span<const uint8_t> in) {
